### ISSUE TITLE
Documentation request: GET requests paths with parameters for api_intro.ipynb

### ISSUE THREAD

**Issue by gwvr** [jupyter/kernel_gateway#201]
I can't grok this notebook (https://github.com/jupyter/kernel_gateway/blob/master/etc/api_examples/api_intro.ipynb) because request examples haven't been included.
I realise that this notebook could conform to some accepted standard, but I wouldn't know where to find the documentation.

I'd be grateful if someone could enlighten me with some example requests to endpoints such as:
* GET /hello/:person
* GET /hello/person
* GET /hello/people
* GET /hello/person

Examples using any/all of webbrowser, curl, python would be useful.

Thanks

**Comment by parente** [comment]
Hi @gwvr. There's some documentation and small examples here http://jupyter-kernel-gateway.readthedocs.io/en/latest/http-mode.html about the parameters. Do those help address some of your questions? If you have any suggestions on how to make them better, we'll certainly welcome pull requests against the doc or example notebooks.

**Comment by gwvr** [comment]
Thanks @parente. I had seen that documentation and it didn't address my questions at all. I'd happily submit a pull request against the notebook if I could grok it in the first place.

I've experimented with running `api_intro.ipynb` using jupyter kernelgateway, and querying it from another notebook.

This works:
```python3
import json
import requests
requests.get(url='http://localhost:8889/hello').content
```
Outputs: `b'hello world\n'`

But this doesn't:
```python3
people = ['Corey', 'Nitin', 'Pete']

data = json.dumps({
    'args': { 
        'person': people
    }
})

requests.get(url='http://localhost:8889/hello/person', data=data).content
```
Outputs: `b"Error KeyError: 'person' \n"`

I don't know where I'm going wrong, hence asking about get request examples.

**Comment by parente** [comment]
The "args" in that endpoint are referring to query parameters which are name/value pairs following a `?` in the URL. So for the endpoint defined as:

```
# GET /hello/person
req = json.loads(REQUEST)
hello_person = req['args']['person'][0]
print(hello_message.format(hello_person))
```

a valid request would be

```
requests.get(url='http://localhost:8889/hello/person?person=Corey').content
```

The doc talks these in different flavors of passing input in the http://jupyter-kernel-gateway.readthedocs.io/en/latest/http-mode.html#getting-the-request-data section. I agree they're too terse. It would be good to update the api_intro notebook so that for each handler, a markdown cell appears above/below it showing how to construct a valid request to it using the requests library like you've done. I can work on that when I get a minute.

**Comment by gwvr** [comment]
Thanks, I had tried a query with `?` in the URL earlier, but I probably got the syntax slightly wrong. I've updated my query notebook and have managed to get requests to the various `Hello` and `People` endpoints working as expected, but I still have issues with requests to many of the other endpoints..

With the `Messages` endpoints, my requests appeared to work but didn't behave as expected. The `put` message didn't seem to be stored for retrieval by a subsequent `get` request.
